core/domain/exp_services.py

```diff
@@ -838,6 +857,14 @@ def _create_exploration(
     # but the creation of an exploration object will fail.
     exploration.validate()
     rights_manager.create_new_exploration_rights(exploration.id, committer_id)
+
+    # Find out all states that need a classifier to be trained.
```

Comment:
Why not combine this logic with the related stuff added to the bottom of this function, instead of splitting up into two separate chunks?